### PR TITLE
Implements delete records

### PR DESCRIPTION
**Pull request by mdulac**
This is a passthrough function to the Admin Client for deleting records

### CODE REVIEW

**Comment by strokyl** [comment]
:+1:

File: src/main/scala/zio/kafka/admin/AdminClient.scala [hunk 1]
```diff
@@ -74,6 +74,14 @@ trait AdminClient {
    */
   def deleteTopic(topic: String): Task[Unit]
 
+  /**
+   * Delete records.
+   */
+  def deleteRecords(
+    recordsToDelete: Map[TopicPartition, RecordsToDelete],
+    deleteRecordsOptions: Option[DeleteRecordsOptions] = None
```
Comment (svroonland) [inline src/main/scala/zio/kafka/admin/AdminClient.scala:82]: Most of the admin methods have a scala variant of such an options object, like `listConsumerGroups` has `ListConsumerGroupsOptions`. Can we implement that here as well?

**Review by svroonland — CHANGES_REQUESTED** [review]
Thanks for the contribution! One request regarding the options parameter

File: src/main/scala/zio/kafka/admin/AdminClient.scala [hunk 2]
```diff
@@ -810,6 +836,13 @@ object AdminClient extends Accessible[AdminClient] {
     }
   }
 
+  case class DeleteRecordsOptions(timeout: Option[Duration]) {
```
Comment (svroonland) [inline src/main/scala/zio/kafka/admin/AdminClient.scala:839]: It's a bit weird to have an `Option[DeleteRecordOptions]` which only has an `Option[Duration]` field, but I guess this matches the java API more closely 👍

**Review by svroonland — APPROVED** [review]
Looks good, thanks!

**Comment by svroonland** [comment]
I see that there's other methods where we're using the java classes directly, raised an issue for that: https://github.com/zio/zio-kafka/issues/418

**Pull request merged**